sormas-backend/src/main/java/de/symeda/sormas/backend/sample/SampleFacadeEjb.java

```diff
@@ -459,6 +459,32 @@ public void validate(SampleDto sample) throws ValidationRuntimeException {
 			throw new ValidationRuntimeException(
 				I18nProperties.getValidationError(Validations.required, I18nProperties.getPrefixCaption(SampleDto.I18N_PREFIX, SampleDto.LAB)));
 		}
+
+		if (checkAssociatedEntities) {
+			validateSampleAssociatedEntities(sample);
+		}
+
+	}
+
+	private void validateSampleAssociatedEntities(SampleDto sample) {
+		if (sample.getAssociatedCase() != null && !caseService.exists(sample.getAssociatedCase().getUuid())) {
+			throw new ValidationRuntimeException(I18nProperties.getValidationError(Validations.validCaseContactOrEventParticipant));
+		}
+		if (sample.getAssociatedContact() != null && !contactService.exists(sample.getAssociatedContact().getUuid())) {
+			throw new ValidationRuntimeException(I18nProperties.getValidationError(Validations.validCaseContactOrEventParticipant));
+		}
+		if (sample.getAssociatedEventParticipant() != null && !eventParticipantService.exists(sample.getAssociatedEventParticipant().getUuid())) {
+			throw new ValidationRuntimeException(I18nProperties.getValidationError(Validations.validCaseContactOrEventParticipant));
+		}
+		if (sample.getLab() != null && !facilityService.exists(sample.getLab().getUuid())) {
+			throw new ValidationRuntimeException(
+				I18nProperties.getValidationError(Validations.required, I18nProperties.getPrefixCaption(SampleDto.I18N_PREFIX, SampleDto.LAB)));
```

Comment:
Please use a different error message here, since the user specified a lab at this point:

"The database does not contain a laboratory with this UUID."